Set up manual page generation

sigv opened the pull request:
For purposes of packaging the tooling, manual pages are a must.

The Makefile build steps have been adjusted to generate manpages using the `help2man` tool which executes the locally built executables with the `--help` and `--version` flags. The Makefile install steps have further been adjusted to place these generated manpages in the /usr/share/man directory by default.

This ensures documentation in manpages does not have to be maintained by hand separated from the codebase, and is instead automatically generated from the latest 'source of truth'.

When building locally without installation steps, you can use the command `man -l rvrrpd.8.gz` to just 'preview' the manpage.

sigv committed:
Set up manual page generation

For purposes of packaging the tooling, manual pages are a must.

The Makefile build steps have been adjusted to generate manpages using
the `help2man` tool which executes the locally built executables with
the `--help` and `--version` flags. The Makefile install steps have
further been adjusted to place these generated manpages in the
/usr/share/man directory by default.

This ensures documentation in manpages does not have to be maintained
by hand separated from the codebase, and is instead automatically
generated from the latest 'source of truth'.

When building locally without installation steps, you can use
the command `man -l rvrrpd.8.gz` to just 'preview' the manpage.

sigv committed:
Ensure manual installation directory and file mode

Just like with the executable, we should ensure that the target
destination directory we are installing to exists. Additionally,
just in case of strange umasks or anything else, we can change
the installed file's mode to ensure the manual is globally
readable, as expected by most.

e3prom commented:
Is `help2man` shipped by default with most distributions or build package set (like build-essentials) ?  If we include the man page generation in the default build process, we will have to mention the dependency in the documentation, otherwise some users may have issues with building the daemon. Alternatively, we can also have the man page generation in a specific Make's target like `doc` and ask the users to manually generate and install the docs using `make doc` and `make doc-install` respectively.

sigv commented:
Fair point - `help2man` is not a default package really, and needs manual installation. I added it by hand to my build dependencies list, but I will provide an update to the documentation as to mention the requirement for it.

I would prefer as to keep the building steps simple (`make && make test && make install`) instead of splitting out a special `make docs`.

sigv committed:
Update documentation for manual generation

It should be noted in documentation, that there is a dependency on
the `help2man` utility -- as this package is not installed by default.

sigv commented on docs/config/install.rst:
```diff
@@ -42,12 +42,13 @@ To build **rVRRPd** from source you must have several programs and libraries ins
    to build the project and its related dependencies (crates).
  * The `OpenSSL <https://www.openssl.org/>`_ development headers
  * The `Netlink Protocol Library Suite <https://www.infradead.org/~tgr/libnl/>`_ development headers *(Linux)*
+ * The `help2man <https://www.gnu.org/software/help2man/>`_ utility, to generate manual pages.
```
Not 100% sure how the syntax on these works with the links, but this appears correct based on the other samples above.

e3prom commented:
Thanks for your update. I merge this PR with the `0.2.0-dev` branch.
